### DISCUSSION TITLE
Is there a way to search if a string present in default branch?

### DISCUSSION THREAD

**Discussion by arunanandhan** [PyGithub/PyGithub#2480] [General]
Is there an api way to search for a string in the current code in a repo?

**Comment by EnricoMi** [comment]
Try

    gh.search_code(...)

See https://docs.github.com/en/rest/reference/search

**Reply by arunanandhan** [reply]
Thanks, I see it searches from all the repositories, is there way to search only for a particular repository?

**Reply by EnricoMi** [reply]
There is an example on that page: https://docs.github.com/en/rest/search#search-code

**Reply by arunanandhan** [reply]
I couldn't find exactly what I am looking for, I'm very sorry, 

can you please paste a code here, if you don't mind?

**Reply by EnricoMi** [reply]
Try (not tested):

    gh.search_code("query repo:user/repo")
    gh.search_code("query", repo="user/repo")

**Reply by arunanandhan** [reply]
Thanks, that works
